waiter/src/waiter/scheduler/kubernetes.clj

```diff
@@ -1252,41 +1255,48 @@
 (defn- start-k8s-watch!
   "Start a thread to continuously update the watch-state atom based on watched K8s events."
   [{:keys [api-server-url scheduler-name watch-state] :as scheduler}
-   {:keys [connect-timeout-ms exit-on-error? resource-key resource-name resource-url socket-timeout-ms
-           streaming-api-request-fn update-fn watch-retries] :as options}]
+   {:keys [connect-timeout-ms exit-on-error? resource-name resource-url socket-timeout-ms
+           streaming-api-request-fn update-fn watch-retries watch-trigger-chan] :as options}]
   (doto
     (Thread.
       (fn k8s-watch []
         (try
           ;; retry getting state updates forever
-          (while true
-            (retry-watch-runner
-              (fn k8s-watch-retried-thunk []
-                (try
-                  (log/info "prepping" resource-name "watch with global sync")
-                  (loop [version (reset-watch-state! scheduler options)
-                         iter 0]
-                    (log/info "starting" resource-name "watch connection, iteration" iter)
-                    (timers/start-stop-time!
-                      (metrics/waiter-timer "scheduler" scheduler-name "state-watch")
-                      (let [request-options (cond-> {}
-                                              connect-timeout-ms (assoc :conn-timeout connect-timeout-ms)
-                                              socket-timeout-ms (assoc :socket-timeout socket-timeout-ms))
-                            watch-url (str resource-url "&watch=true&resourceVersion=" version)]
-                        ;; process updates forever (unless there's an exception)
-                        (doseq [json-object (streaming-api-request-fn resource-name watch-url request-options)]
-                          (when json-object
-                            (if (= "ERROR" (:type json-object))
-                              (log/info "received error update in watch" resource-name json-object)
-                              (update-fn json-object))))))
-                    ;; when the watch connection closed normally (i.e., no HTTP error code response),
-                    ;; retry the watch (before falling back to the global query again) `watch-retries` times.
-                    (when (< iter watch-retries)
-                      (when-let [version' (latest-watch-state-version scheduler options)]
-                        (recur version' (inc iter)))))
-                  (catch Exception e
-                    (log/warn "error in" resource-name "state watch thread:" e)
-                    (throw e))))))
+          (let [resource-name-lower (str/lower-case resource-name)]
+            (while true
+              (retry-watch-runner
+                (fn k8s-watch-retried-thunk []
```

Comment:
What is a thunk?

Reply:
A thunk is a function that takes no argument. 

This didnt necessarily change in this PR, using whitespace off in diff might help.